tree/dataframe/test/dataframe_vary.cxx

```diff
@@ -767,19 +1169,55 @@ TEST_P(RDFVary, VaryTake)
    EXPECT_EQ(sorted(rs["x:1"]), std::vector<int>({1, 2, 3}));
 }
 
-TEST_P(RDFVary, DISABLED_VaryForeach)
-{
+class Product {
+public:
+   Product() : _x(0), _y(0) {}
+   Product(double x, double y) : _x(x), _y(y) {}
+   ~Product() {}
 
-}
+   double GetProduct() { return _x * _y; }
 
-TEST_P(RDFVary, DISABLED_VarySnapshot)
-{
+private:
+   double _x, _y;
+};
+
+TEST_P(RDFVary, DISABLED_VaryForeach) {}
 
+TEST_P(RDFVary, VarySnapshot)
+{
+   const auto fname = "dummy.root";
+   auto h = ROOT::RDataFrame(10)
+               .Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"})
+               .Vary(
+                  "x",
+                  [](int x) {
+                     return ROOT::RVecI{x - 1, x + 1};
+                  },
+                  {"x"}, 2)
+               .Snapshot<int>("t", fname, {"x"});
+   EXPECT_THROW(VariationsFor(h), std::logic_error); // not (yet) implemented
 }
 
-TEST_P(RDFVary, DISABLED_VaryOnPartialResult)
+TEST_P(RDFVary, VaryOnPartialResult)
 {
+   auto h = ROOT::RDataFrame(10)
+               .Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"})
+               .Vary(
+                  "x",
+                  [](int x) {
+                     return ROOT::RVecI{x - 1, x + 1};
+                  },
+                  {"x"}, 2)
+               .Sum<int>("x");
+   std::atomic_uint counter(0u);
+   h.OnPartialResult(h.kOnce, [&counter](ULong64_t) { counter++; });
+   auto hs = VariationsFor(h);
 
+   EXPECT_EQ(*h, 45);
+   EXPECT_EQ(hs["nominal"], 45);
+   EXPECT_EQ(hs["x:0"], 35);
+   EXPECT_EQ(hs["x:1"], 55);
+   EXPECT_EQ(counter, 1u);
```

Comment:
This was just testing `OnPartialResult` on the nominal result, but that is already tested.

I don't know how OnPartialResult should interact with `Vary`, no need for a test for now.

```suggestion
```

Reply:
I only disabled it so that we keep it in mind.

Reply:
disabling the test code suggests that one day we want the test to pass, but that's not the case here: we don't even know how the test code should look like

Reply:
test removed in the new commit.